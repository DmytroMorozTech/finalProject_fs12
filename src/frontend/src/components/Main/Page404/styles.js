import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({
  container: {
    width: '100vw',
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: theme.palette.background.default,
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 150
  },

  header: {
    backgroundColor: theme.palette.background.paper,
    width: '100vw',
    paddingLeft: '15vw'
  },

  pageImg: {
    marginTop: '15vh',
    backgroundAttachment: 'center',
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    width: '300px'
  },

  link: {
    textDecoration: 'none'
  }
}))
