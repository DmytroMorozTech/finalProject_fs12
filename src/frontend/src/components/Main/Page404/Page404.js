import React from 'react'
import styles from './styles'
import imgPage from '../../../temporaryImages/page404img.png'
import SharedButton from '../../../shared/Button/SharedButton'
import { Link } from 'react-router-dom'
import LinkedinLogo from '../../../shared/LinkedinLogo/LinkedinLogo'

function Page404 () {
  const classes = styles()

  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <LinkedinLogo/>
      </div>
      <img alt="page 404" src={imgPage} className={classes.pageImg}/>
      <h1>This page doesn't exist</h1>
      <h2>Please check your URL or return to LinkedIn home.</h2>
      <Link exact to='/home' className={classes.link}>
        <SharedButton title="Go to your feed" variant="outlined"/>
      </Link>
    </div>
  )
}

export default Page404
