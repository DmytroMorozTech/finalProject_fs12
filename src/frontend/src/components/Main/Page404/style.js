import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(() => ({
   text: {
    color: 'red',
    fontSize: '16px'
   }
}))