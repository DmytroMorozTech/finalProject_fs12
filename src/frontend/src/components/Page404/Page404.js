import React from 'react'
import styles from './styles'
import SharedButton from '../../shared/SharedButton/SharedButton'
import { Link } from 'react-router-dom'
import LinkedinLogo from '../../shared/LinkedinLogo/LinkedinLogo'
import Image from '../../shared/Image/Image'

function Page404 () {
  const classes = styles()

  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <Link exact to='/home' className={classes.link}>
          <LinkedinLogo/>
        </Link>
      </div>
      <Image
        imageUrl={'linkedin/general/jrn3axnlbe8gxborjqxb'}
        alt={'page 404'}
        className={classes.pageImg}
      />
      <h1>This page doesn't exist</h1>
      <h2 className={classes.text}>Please check your URL or return to LinkedIn home.</h2>
      <Link exact to='/home' className={classes.link}>
        <SharedButton title="Go to your feed" variant="outlined"/>
      </Link>
    </div>
  )
}

export default Page404
