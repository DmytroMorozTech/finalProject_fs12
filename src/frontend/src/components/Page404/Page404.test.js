import { shallow } from 'enzyme'
import React from 'react'
import Page404 from './Page404'

const setUp = () => shallow(<Page404 />)

describe('should render Page404 component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
