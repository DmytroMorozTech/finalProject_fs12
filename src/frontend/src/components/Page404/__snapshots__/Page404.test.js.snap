// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page404 component to match snapshot 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-header-2"
  >
    <Link
      className="makeStyles-link-4"
      exact={true}
      to="/home"
    >
      <LinkedinLogo />
    </Link>
  </div>
  <Image
    alt="page 404"
    className="makeStyles-pageImg-3"
    imageUrl="linkedin/general/jrn3axnlbe8gxborjqxb"
  />
  <h1>
    This page doesn't exist
  </h1>
  <h2
    className="makeStyles-text-5"
  >
    Please check your URL or return to LinkedIn home.
  </h2>
  <Link
    className="makeStyles-link-4"
    exact={true}
    to="/home"
  >
    <SharedButton
      title="Go to your feed"
      variant="outlined"
    />
  </Link>
</div>
`;
